### DISCUSSION TITLE
Trying to get debug information from strongSwan 5.9.5

### DISCUSSION THREAD

**Discussion by ecomm1** [strongswan/strongswan#1532] [General]
Hi experts,

We are using (strongSwan 5.9.5, Linux 5.15.0-56-generic, x86_64 Ubuntu) we have a VPN using Strongswan which is not showing the debug information so we can't see if it's working or not.

We have followed these steps to add debugging https://wiki.strongswan.org/projects/strongswan/wiki/LoggerConfiguration/24
and https://docs.strongswan.org/docs/5.9/config/logging.html

I added to /etc/strongswan.conf
    filelog {
        charon {
            # path to the log file, specify this as section name in versions prior to 5.7.0
            path = /var/log/charon.log
            # add a timestamp prefix
            time_format = %b %e %T
            # prepend connection name, simplifies grepping
            ike_name = ciscoasa
            # overwrite existing files
            append = no
            # increase default loglevel for all daemon subsystems
            default = 2
            # flush each line to disk
            flush_line = yes
        }
        stderr {
            # more detailed loglevel for a specific subsystem, overriding the
            # default loglevel.
            ike = 2
            knl = 3
        }
    }
    # and two loggers using syslog
    syslog {
        # prefix for each log message
        identifier = charon-custom
        # use default settings to log to the LOG_DAEMON facility
        daemon {
        }
        # very minimalistic IKE auditing logs to LOG_AUTHPRIV
        auth {
            default = -1
            ike = 0
        }
    }

I added to /etc/ipsec.conf
config setup
        # strictcrlpolicy=yes
        # uniqueids =
        #
        charondebug="ike 2, knl 3, cfg 0"

In the syslog we are now getting these errors:
Feb  5 11:50:22  systemd[1]: fwupd.service: Consumed 1.181s CPU time.
Feb  5 11:50:56  kernel: [566368.718727] audit: type=1400 audit(1675590656.662:58): apparmor="DENIED" operation="open" profile="/usr/lib/ipsec/charon" name="/var/log/charon.log" pid=202149 comm="charon" requested_mask="wc" denied_mask="wc" fsuid=0 ouid=0
Feb  5 11:50:56  kernel: [566368.718788] audit: type=1400 audit(1675590656.662:59): apparmor="DENIED" operation="mknod" profile="/usr/lib/ipsec/charon" name="/root/syslog" pid=202149 comm="charon" requested_mask="c" denied_mask="c" fsuid=0 ouid=0
Feb  5 11:51:55  kernel: [566427.809247] audit: type=1400 audit(1675590715.749:60): apparmor="DENIED" operation="mknod" profile="/usr/lib/ipsec/charon" name="/var/log/charon.log" pid=202188 comm="charon" requested_mask="c" denied_mask="c" fsuid=0 ouid=0
Feb  5 11:51:55  kernel: [566427.809303] audit: type=1400 audit(1675590715.749:61): apparmor="DENIED" operation="mknod" profile="/usr/lib/ipsec/charon" name="/root/syslog" pid=202188 comm="charon" requested_mask="c" denied_mask="c" fsuid=0 ouid=0

Any idea how to get this to work?
/var/log/syslog is getting the errors as shown above, but the file at /var/log/charon.log is not being created.

Thanks any help appreciated.

**Comment by ecomm1** [comment]
update

**Comment by tobiasbrunner** [comment]
> Any idea how to get this to work?

As described e.g. [here](https://docs.strongswan.org/docs/5.9/support/helpRequests.html#_dealing_with_linux_security_modules_lsm), you have to consider LSMs like AppArmor when configuring arbitrary log file paths.
